src/fm/cp_fm_diag.F

```diff
@@ -882,6 +899,140 @@ SUBROUTINE cp_fm_syevx(matrix, eigenvectors, eigenvalues, neig, work_syevx)
 
    END SUBROUTINE cp_fm_syevx
 
+! **************************************************************************************************
+!> \brief decomposes a quadratic matrix into its singular value decomposition
+!> \param matrix_a ...
+!> \param matrix_eigvl ...
+!> \param matrix_eigvr_t ...
+!> \param eigval ...
+!> \param info ...
+!> \author Maximilian Graml
+! **************************************************************************************************
+   SUBROUTINE cp_fm_svd(matrix_a, matrix_eigvl, matrix_eigvr_t, eigval, info)
+
+      TYPE(cp_fm_type), INTENT(IN)              :: matrix_a
+      TYPE(cp_fm_type), INTENT(OUT)             :: matrix_eigvl, matrix_eigvr_t
```

Comment:
I am not sure whether other diagonalization/linear algebra routines in the fm package create matrices inside (without the obvious exceptions). I would rather push the calls to cp_fm_create outside of this routine to make the allocation step explicit. This makes other developers aware that the calling routing is responsible to release the matrix later.